tasks/restore_config.rb

```diff
@@ -0,0 +1,55 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require_relative '../lib/puppet/util/task_helper'
+task = Puppet::Util::TaskHelper.new('cisco_ios')
+
+result = {}
+
+unless Puppet.settings.global_defaults_initialized?
+  Puppet.initialize_settings
+end
+
+VALID_CISCO_HEADERS = ['show running-config', 'Building configuration', 'Current configuration']
+
+def config_to_restore(raw_config_output)
+  header_match_count = 0
+  line = 0
+  start_line = 0
+  end_line = 0
+
+  raw_config_output.each do |rco|
+    header_match_count += 1 if VALID_CISCO_HEADERS.select { |vch| rco.start_with? vch }.size == 1
+    start_line = (line + 1) if (header_match_count == 3) && (rco.start_with? 'version')
+    end_line = line if rco.start_with? 'end'
+    break if start_line.positive? && end_line.positive?
+    # If we haven't found the headers by line 20, let's exit as this doesn't seem to be a valid Cisco
+    # backup config retrieved using 'show running-config'
+    raise 'Could not detect expected headers from Cisco backup' if line >= 20 && header_match_count < 3
```

Comment:
On error messages always specify what the code was looking for, e.g.

```suggestion
    raise 'Could not detect the expected headers ("foo", "bar") from Cisco backup within the first 20 lines' if line >= 20 && header_match_count < 3
```